lib/openstax/aws/template.rb

```diff
@@ -84,6 +87,12 @@ def upload_once
 
   protected
 
+    def initialize(absolute_file_path: nil, body: nil)
+      raise "One of `absolute_file_path` or `body` must be set" if absolute_file_path.blank? && body.nil?
+      @absolute_file_path = absolute_file_path
+      @body = body.try(:dup)
```

Comment:
is the idea here that @body can be nil?  if not, try! to raise exception, or better use the ruby 2.3+ safe navigation operator.

Reply:
@body can be nil.  Started with `&.` but didn't want to force 2.3+ in case someone has a lower 2 version.